cxx-squid/src/main/java/org/sonar/cxx/preprocessor/ExpressionEvaluator.java

```diff
@@ -144,7 +145,8 @@ private BigInteger evalComplexAst(AstNode exprAst) {
     } else if ("functionlikeMacro".equals(nodeType)) {
       return evalFunctionlikeMacro(exprAst);
     } else {
-      throw new EvaluationException("Unknown expression type '" + nodeType + "'");
+      LOG.error("'evalComplexAst' Unknown expression type '" + nodeType + "' for AstExt '" + exprAst.getToken() + "', assuming 0");
+      return BigInteger.ZERO;
```

Comment:
its this is what i understand also, i get already some of those in my logs (expression something). i dont know if throwing exception or return .ZERO is the same, at least the processor does a lot of not found convert to 0. Anyway, the error  in this way will be better understandable, the stacktrace imo brings no additional information. Also the print of the stack trace is not done inside our code but in the sslr api so i am unable to disable the trace. I supose if the integration tests are ok, this change should not affect users.